src/system/mod.rs

```diff
@@ -960,4 +966,17 @@ mod tests {
         let (_, status) = child_pid.wait(WaitOptions::new()).unwrap();
         assert_eq!(status.exit_status(), Some(0));
     }
+
+    #[cfg(target_os = "linux")]
+    #[test]
+    fn proc_stat_test() {
+        use super::{read_proc_stat, Process, WithProcess::Current};
+        // the process is 'sleeping' (apparently)
+        assert_eq!("S", read_proc_stat::<String>(Current, 2).unwrap());
+        let parent = Process::parent_id().unwrap();
+        // field 3 is always the parent process
+        assert_eq!(parent, read_proc_stat::<i32>(Current, 3).unwrap());
+        // this next field should always be 0 (which precedes an important bit of info for us!)
+        assert_eq!(0, read_proc_stat::<i32>(Current, 20).unwrap());
```

Comment:
Maybe also add a test with a fixed stat string to parse?

Reply:
That is possible, but would necessitate further refactors of `read_proc_stat` to split out the string-component. I think this test (in particular the test for the "zero" and "parent pid") is sufficient to detect an indexing-based error in the future.